### ISSUE TITLE
Launch Tensorboard in user's namespace in multi-user mode

### ISSUE THREAD

**Issue by IronPan** [kubeflow/pipelines#3294]
Part of #1223

**Comment by Bobgy** [comment]
I quickly verified if this works e2e and found something broken in viewer controller when a viewer CR is created in user namespace.
Not finding much error message in viewer controller, so I need some extra for figuring out what went wrong.

**Comment by Bobgy** [comment]
@jingzhang36 @neuromage Does current viewer controller support viewer CRs in multiple namespaces?

**Comment by Bobgy** [comment]
Nevermind, I figured out viewer controller has a namespace env which controls which namespace it works on, if the env is empty string, then it watches for all namespaces.

**Comment by Bobgy** [comment]
Changes to support this:
* [x] Manifest change to let viewer controller watch all namespaces: https://github.com/Bobgy/manifests/commit/de52914d89b1f3a16b4842d0fd99ab7dd2fafd44
* [ ] UI changes to specify namespace in viewer CR

**Comment by Bobgy** [comment]
I tried enabling istio sidecars for tensorboard services, but istio rbac still reject all requests when user should be authorized.
After some troubleshooting, I found that it seems istio didn't automatically find out the traffic is in http, so authorization rules based on http properties didn't take effect. https://istio.io/docs/ops/configuration/traffic-management/protocol-selection/

Therefore, the required if is to add a name to tensorboard service port.
